dp/core/syscall.c

```diff
@@ -199,7 +199,8 @@ static int sys_bpoll(struct bsys_desc __user *d, unsigned int nr)
 				KSTATS_PUSH(idle, NULL);
 
 				start = rdtsc();
-				eth_rx_idle_wait(deadline);
+				if (percpu_get(eth_num_queues) > 0)
```

Comment:
So the idea would be that we check only the queues belonging to the current cpu, using `percpu_get(eth_rxqs[i])` and they could be from different devices?